workspaces/util/src/__tests__/swap-test.ts

```diff
@@ -0,0 +1,66 @@
+import { describe, expect, it } from "@jest/globals";
+
+import { swap } from "../swap";
+
+describe("swap", () => {
+  it("swaps two elements in an array of numbers", () => {
+    const arr = [1, 2, 3, 4];
+    swap(arr, 1, 3);
+    expect(arr).toStrictEqual([1, 4, 3, 2]);
+  });
+
+  it("swaps two elements in an array of strings", () => {
+    const arr = ["a", "b", "c", "d"];
+    swap(arr, 0, 2);
+    expect(arr).toStrictEqual(["c", "b", "a", "d"]);
+  });
+
+  it("swaps two elements in an array of mixed types", () => {
+    const arr = [1, "a", true, 3.14];
+    swap(arr, 1, 3);
+    expect(arr).toStrictEqual([1, 3.14, true, "a"]);
+  });
+
+  it("does nothing when swapping the same index", () => {
+    const arr = [1, 2, 3];
+    swap(arr, 1, 1);
+    expect(arr).toStrictEqual([1, 2, 3]);
+  });
+
+  it("handles arrays with one element", () => {
+    const arr = [42];
+    swap(arr, 0, 0);
+    expect(arr).toStrictEqual([42]);
+  });
+
+  it("handles arrays with multiple elements when swapping first and last indices", () => {
```

Comment:
I'm not sure I follow the idea behind this test. Can you say more?

Reply:
It was my first thought of arrays with multiple elements but then I came up with new idea with large arrays. Forgot to delete it